fractal-puppy/Engine
# Resource Manager Improvements

Two main improvements:
- PanelBrowser: You can choose between different load configurations for Resources using the PanelBrowser. At this moment only ResourceAudio has load configuration. You can delete folders if they are empty.

- PanelResourceManager: The resources list isn't refreshed automatically to improve efficiency, you can turn auto refresh on again from the menu bar. You can filter the resources list to only show the desired type of resource or the ones loaded in memory. Added ResourceAudio window.

To test (1): 
- Using the PanelBrowser go to "Assets/Audio"
- Right-click on any audio file and select "Load Settings"
- Check "Streamed" checkbox
- Close the engine and open it again
- Using the PanelBrowser go to "Assets/Audio"
- Right-click on the same audio file and select "Load Settings"
- See that the checkbox is still checked.

To test (2):
- Using the PanelBrowser right click and choose "New Folder"
- Give the folder a name
- Right-click on the folder and choose "Delete"
- Click on yes
- Check that the folder has been deleted

To test (3):
- Open the ResourceManager tool using the menu bar
- Go to "Resource->Filter by->Resource type" using the menu bar and choose any resource type
- Check that the list has been updated to show only that type of resource
https://gyazo.com/290cd30e18ad083a3e06c58b81fc5098

To test (4):
- Open the ResourceManager tool using the menu bar
- Go to "Resource->Filter by->Reference count" using the menu bar and choose an option
- Check that the list has been updated to show only either not loaded resources or the ones loaded
